Tests/GRPCInterceptorsTests/TracingInterceptorTests.swift

```diff
@@ -134,41 +234,200 @@ final class TracingInterceptorTests: XCTestCase {
+    let interceptor = ClientOTelTracingInterceptor(
+      serverHostname: "someserver.com",
+      networkTransportMethod: "tcp"
+    )
+    let (_, continuation) = AsyncStream<String>.makeStream()
+    serviceContext.traceID = traceIDString
+
+    // FIXME: use 'ServiceContext.withValue(serviceContext)'
+    //
+    // This is blocked on: https://github.com/apple/swift-service-context/pull/46
+    await ServiceContext.$current.withValue(serviceContext) {
+      let methodDescriptor = MethodDescriptor(
+        fullyQualifiedService: "TracingInterceptorTests",
+        method: "testClientInterceptorErrorEncountered"
+      )
+      do {
+        _ = try await interceptor.intercept(
+          request: .init(producer: { writer in
+            try await writer.write("request")
+            throw TracingInterceptorTestError.testError
+          }),
+          context: self.getClientContext(forMethod: methodDescriptor)
+        ) { stream, _ in
+          // Assert the metadata contains the injected context key-value.
+          XCTAssertEqual(stream.metadata, ["trace-id": "\(traceIDString)"])
+
+          // Write into the response stream to make sure the `producer` closure's called.
+          let writer = RPCWriter(wrapping: TestWriter(streamContinuation: continuation))
+          try await stream.producer(writer)
+          continuation.finish()
+
+          return .init(
+            metadata: [],
+            bodyParts: RPCAsyncSequence(
+              wrapping: AsyncThrowingStream<StreamingClientResponse.Contents.BodyPart, any Error> {
+                $0.yield(.message(["response"]))
+                $0.finish()
+              }
+            )
+          )
+        }
+        XCTFail("Should have thrown")
+      } catch {
+        AssertTestSpanComponents(forMethod: methodDescriptor) { events in
+          // We errored when writing our request, so this is the only event we should be seeing
+          XCTAssertEqual(events.map({ $0.name }), ["Request started"])
+        } assertAttributes: { attributes in
+          // The attributes should not contain a grpc status code, as the request was never even sent.
+          XCTAssertEqual(
+            attributes,
+            [
+              "rpc.system": .string("grpc"),
+              "rpc.method": .string(methodDescriptor.method),
+              "rpc.service": .string(methodDescriptor.service.fullyQualifiedService),
+              "server.address": .string("someserver.com"),
+              "server.port": .int(567),
+              "network.peer.address": .string("10.1.2.80"),
+              "network.peer.port": .int(567),
+              "network.transport": .string("tcp"),
+              "network.type": .string("ipv4"),
+            ]
+          )
+        } assertStatus: { status in
+          XCTAssertNil(status)
+        } assertErrors: { errors in
+          XCTAssertEqual(errors, [.testError])
+        }
+      }
+    }
+  }
+
+  func testClientInterceptorErrorReponse() async throws {
```

Comment:
We should also test a response which is accepted (i.e. `response.accepted` is `success`) but ends with a non-zero status.